model/task_queue.go

```diff
@@ -81,6 +81,10 @@ func (self *TaskQueue) Save() error {
 }
 
 func (self *TaskQueue) FindTask(spec TaskSpec) *TaskQueueItem {
+	if spec.GroupName == "" {
```

Comment:
Doesn't this break for tasks not in a group?

Reply:
I'm making changes so that tasks not in a group get the same group name as their id. However, for the moment I haven't yet done those changes, so I don't know if this is safe.

Reply:
The way this gets accessed, is via the MatchingOrNextTask method in model, which first sees if there is a matching task for the task group and returns that. If there aren't any, then it falls back to returning the next task.